### PR TITLE
cgroups: add support for oom control

### PR DESCRIPTION
**Pull request by HuKeping**
Redo #411

### CODE REVIEW

File: cgroups/fs/memory.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func (s *MemoryGroup) Set(d *data) error {
 	}()
 
 	// Only set values if some config was specified.
-	if d.c.Memory != 0 || d.c.MemoryReservation != 0 || d.c.MemorySwap != 0 {
+	if d.c.Memory != 0 || d.c.MemoryReservation != 0 || d.c.MemorySwap != 0 || d.c.OomKillDisable {
```
Comment (vmarmol) [inline cgroups/fs/memory.go]: Break out OomKillDisable into its own if statement. It shouldn't be bundled with these here (neither should Reservation and Memory for that matter, but that's out of scope :) ).

File: configs/cgroup.go [hunk 2]
```diff
@@ -51,4 +51,7 @@ type Cgroup struct {
 
 	// Parent slice to use for systemd TODO: remove in favor or parent
 	Slice string `json:"slice"`
+
+	// Disable OOM Killer
```
Comment (vmarmol) [inline configs/cgroup.go]: nit: Whether to...

**Comment by HuKeping** [comment]
why every time i redo the commit it will auto close the PR? Here's my steps:

$ git reset --soft HEAD^             (1)

$ edit                                      (2)

$ git commit -a -c ORIG_HEAD  (3)

$ git push -f 		              (4)

**Comment by LK4D4** [comment]
@HuKeping I'm using `--amend` or `rebase -i` + `--amend` if I need to change not last commit. Seems like `github` have some protection against your case.

**Comment by rjnagal** [comment]
I think your way makes the original commit id go away, which will close the
PR. Doing a rebase -i will preserve the id and github won't bother you.

On Thu, Feb 26, 2015 at 6:16 PM, Alexander Morozov <notifications@github.com

**Comment by HuKeping** [comment]
get it,thanks

**Pull request closed without merging**